src/inference/mh.jl

```diff
@@ -187,34 +201,77 @@ function AbstractMCMC.sample_init!(
 
     # Get `init_theta`
     initialize_parameters!(spl; verbose=verbose, kwargs...)
+
+    # If we're doing random walk with a covariance matrix,
+    # just link everything before sampling.
+    if spl.alg.proposals isa AdvancedMH.RandomWalkProposal{<:MvNormal}
+        link!(spl.state.vi, spl)
+    end
```

Comment:
Maybe also write this more generically as
```julia
maybe_link!(spl.state.vi, spl, spl.alg.proposals)
```
and define
```julia
maybe_link!(varinfo, sampler, proposal) = nothing
maybe_link!(varinfo, sampler, proposal::AdvancedMH.RandomWalkProposal) = link!(varinfo, sampler)
```
?

Reply:
I think that's more complex than it needs to be -- I'd rather stick closely to convention (used elsewhere) so it's immediately visible when something is being linked.

Reply:
I see that point, it's just that IMO hardcoding type checks in the general definitions of `sample_init!`, `step!` etc makes it more difficult to extend them and to achieve similar behaviour for different proposals. Maybe one could choose a name that's less focused on transforming variables and more something like a proposal-specific `sample_init!`, `step!`, and `sample_end!`, then it would be easy to define optimized behaviour depending on the type of proposal.

Reply:
Okay, I added `maybe_link!` and `maybe_invlink!`.